lib/wos_queries.rb

```diff
@@ -0,0 +1,279 @@
+
+# Queries on the Web of Science (or Web of Knowledge)
+class WosQueries
+
+  # this is the maximum number that can be returned in a single query by WoS
+  MAX_RECORDS = 100
+
+  QUERY_LANGUAGE = 'en'.freeze
+
+  # limit the start date when searching for publications, format: YYYY-MM-DD
+  START_DATE = '1970-01-01'.freeze
+
+  attr_reader :wos_client
+
+  # @param wos_client [WosClient] a Web Of Science client
+  # @param database [String] a WOS database identifier (default 'WOK')
+  def initialize(wos_client, database = 'WOK')
+    @wos_client = wos_client
+    @database = database
+  end
+
+  # @param uid [String] a WOS UID
+  # @return [WosRecords]
+  def cited_references(uid)
+    cited_references_collator(uid)
+  end
+
+  # @param uid [String] a WOS UID
+  # @return [WosRecords]
+  def citing_articles(uid)
+    citing_articles_collator(uid)
+  end
+
+  # @param uid [String] a WOS UID
+  # @return [WosRecords]
+  def related_records(uid)
+    related_records_collator(uid)
+  end
+
+  # @param uids [Array<String>] a list of WOS UIDs
+  # @return [WosRecords]
+  def retrieve_by_id(uids)
+    retrieve_by_id_collator(uids)
+  end
+
+  # @param name [String] a CSV name pattern: {last name}, {first_name} [{middle_name} | {middle initial}]
+  # @return [WosRecords]
+  def search_by_name(name)
```

Comment:
Why do you have these 5 public methods that are effectively aliases for private ones?  Put the body of the private methods here and remove the "collator" methods.

Reply:
I'll try to move some stuff where this is done in #242 already, into this PR.  This pattern of using a collator emerged from doing one request first and then another and another and then noticing that the collators themselves could be refactored, until they became so small that they, well, don't need to exist any more.

Reply:
Done